app/models/container_image.rb

```diff
@@ -11,6 +11,7 @@ class ContainerImage < ApplicationRecord
   DOCKER_PULLABLE_PREFIX = "docker-pullable://".freeze
   DOCKER_PREFIXES = [DOCKER_IMAGE_PREFIX, DOCKER_PULLABLE_PREFIX].freeze
 
+  belongs_to :parent, :polymorphic => true
```

Comment:
I didn't mean to create a genealogy structure here, Just to have all container image types inherit from the same base so that they can be addressed the same and share code.